modules/artifact/src/test/java/org/mule/runtime/module/artifact/classloader/ResourceReleaserTestCase.java

```diff
@@ -171,6 +178,36 @@ public void createsInstanceOnlyOnce() throws IOException {
     }
   }
 
+  @Test
+  public void stopOracleDriverLeakyThreads() throws Exception {
+    JdbcResourceReleaser resourceReleaser = new JdbcResourceReleaser();
+
+    Method killOracleThreadsMethod = resourceReleaser.getClass().getDeclaredMethod("disposeOracleDriverThreads");
+    killOracleThreadsMethod.setAccessible(true);
+
+    Timer timerThread = new Timer();
+    TimerTask task = new TimerTask() {
+
+      @Override
+      public void run() {
+        System.out.println("Timer");
+      }
+    };
+
+    timerThread.schedule(task, 10, 100000);
+
+    Thread[] threads = new Thread[activeCount()];
+    enumerate(threads);
+
+    killOracleThreadsMethod.invoke(resourceReleaser);
+
+    for (Thread thread : threads) {
+      if (thread.getName().contains(ORACLE_DRIVER_TIMER_THREAD_NAME)) {
+        check(15000, 100, () -> TERMINATED.equals(thread.getState()));
```

Comment:
this is testing nothing:
1) You're not testing the classloading navigation part. This doesn't give any indication on wether this actually works with the DB connector
2) You're starting a timer, but at no point you gave it the correct name
3) You're not considering the possibility of the thread not being found and the check() never taking place.